src/stream/stream/mod.rs

```diff
@@ -1147,6 +1149,45 @@ extension_trait! {
         {
             FromStream::from_stream(self)
         }
+
+        #[doc = r#"
+            Lexicographically compares the elements of this `Stream` with those
+            of another using 'Ord'. 
+            
+            # Examples
+            ```
+            # fn main() { async_std::task::block_on(async {
+            #
+            use async_std::prelude::*;
+            use std::collections::VecDeque;
+            
+            use std::cmp::Ordering;
+            let s1 = VecDeque::from(vec![1]);
+            let s2 = VecDeque::from(vec![1, 2]);
+            let s3 = VecDeque::from(vec![1, 2, 3]);
+            let s4 = VecDeque::from(vec![1, 2, 4]);
+
+            assert_eq!(s1.clone().cmp(s1.clone()).await, Ordering::Equal);
+            assert_eq!(s1.clone().cmp(s2.clone()).await, Ordering::Less);
+            assert_eq!(s2.clone().cmp(s1.clone()).await, Ordering::Greater);       
+            assert_eq!(s3.clone().cmp(s4.clone()).await, Ordering::Less);
+            assert_eq!(s4.clone().cmp(s3.clone()).await, Ordering::Greater);  
+
+            #
+            # }) }
+            ```
+        "#]
+        fn cmp<S>(
+           self,
+           other: S
+        ) -> impl Future<Output = Ordering> [CmpFuture<Self, S>]
+        where
+            Self: Sized + Stream,
```

Comment:
Is `Self: Stream` necessary at all?

Reply:
When I remove `+ Stream` from this clause , it builds and passes tests but I get the following error from rustdoc: 
```
error[E0277]: the trait bound `Self: stream::stream::Stream` is not satisfied
    --> src/stream/stream/mod.rs:1180:9
     |
1180 | /         fn cmp<S>(
1181 | |            self,
1182 | |            other: S
1183 | |         ) -> impl Future<Output = Ordering> [CmpFuture<Self, S>]
...    |
1189 | |             CmpFuture::new(self, other)
1190 | |         }
     | |_________^ the trait `stream::stream::Stream` is not implemented for `Self`
     |
     = help: consider adding a `where Self: stream::stream::Stream` bound
```